controllers/useraccount/useraccount_controller.go

```diff
@@ -675,3 +778,61 @@ func getUsersByOwnerName(ctx context.Context, cl client.Client, owner string) ([
 	}
 	return userList.Items, nil
 }
+
+func getConfigMapByLabel(ctx context.Context, cl client.Client, uid string) ([]corev1.ConfigMap, error) {
+	configMapList := &corev1.ConfigMapList{}
+	labels := map[string]string{ConsoleUserSettingsIdentifier: "true", ConsoleUserSettingsUID: uid}
+	err := cl.List(ctx, configMapList, client.MatchingLabels(labels), client.InNamespace(UserSettingNS))
+	if err != nil {
+		return []corev1.ConfigMap{}, err
+	}
+	return configMapList.Items, nil
+}
+
+func getRolesByLabel(ctx context.Context, cl client.Client, uid string) ([]v1.Role, error) {
+	roleList := &v1.RoleList{}
+	labels := map[string]string{ConsoleUserSettingsIdentifier: "true", ConsoleUserSettingsUID: uid}
+	err := cl.List(ctx, roleList, client.MatchingLabels(labels), client.InNamespace(UserSettingNS))
+	if err != nil {
+		return []v1.Role{}, err
+	}
+	return roleList.Items, nil
+}
+
+func getRoleBindingsByLabel(ctx context.Context, cl client.Client, uid string) ([]v1.RoleBinding, error) {
+	rbList := &v1.RoleBindingList{}
+	labels := map[string]string{ConsoleUserSettingsIdentifier: "true", ConsoleUserSettingsUID: uid}
+	err := cl.List(ctx, rbList, client.MatchingLabels(labels), client.InNamespace(UserSettingNS))
+	if err != nil {
+		return []v1.RoleBinding{}, err
+	}
+	return rbList.Items, nil
+}
+
+func getObjectByName(ctx context.Context, cl client.Client, name string, object client.Object) (client.Object, error) {
```

Comment:
Let's return `nil, nil` if the object not found (in this case the client will return a NotFound error which you just ignore). And `nil, err` if there is another error (not NotFound). Also add a description to the function explaining it.

Also let's rename it to something like `getConsoleUserSettingObjectByName()` or something. So it's clear that it's specific to the console user settings.
And maybe let's move all console user setting specific functions like getXXXObjects, delete...() and the constants to a different file? `usersettings.go` within the same package? So we don't have too big useraccount_controller.go file?